src/executor.rs

```diff
@@ -74,9 +77,14 @@ impl Executor {
             sleep_params: SleepParams::default(),
             retention_mode: RetentionMode::RemoveFinished,
             task_type: None,
+            shared_state: None,
```

Comment:
I think shared_state can be `None`? Maybe it can be passed to the new method?

Reply:
Not sure what the best choice here is. Should SharedState be a required dependency of Executor? Or should Executor function without a SharedState set?
If it's required, we could add it to `new` and create a helper method like `new_without_shared_state` for tests that don't care about SharedState in Executor. That method could use the default value `Arc::new(RwLock::new(WorkerState::Running))`.
If SharedState *isn't* required, we could keep the code the same (using a setter), or create helper method `new_with_shared_state` to include state.

Which do you think is better?

Reply:
I think your current approach is good enough